domain/annotation/service/service.go

```diff
@@ -0,0 +1,51 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/core/annotations"
+)
+
+// State describes retrieval and persistence methods for annotations.
+type State interface {
+	// GetAnnotations retrieves all the annotations associated with a given ID.
+	// If no annotations are found, an empty map is returned.
+	GetAnnotations(ctx context.Context, ID annotations.ID) (map[string]string, error)
+
+	// UpdateAnnotations associates key/value annotation pairs with a given ID.
+	// If annotation already exists for the given ID, then it will be updated with
+	// the given value.
+	UpdateAnnotations(ctx context.Context, ID annotations.ID, annotations map[string]string) error
```

Comment:
Quick question, will probably require digging into the old annotation code... How do you unset/remove a value. If it's just replace all, the @tlm is incorrect and this should be set and not update.

I believe annotation is just key value pairs. Otherwise we would need remove annotations.

Reply:
See: https://github.com/juju/juju/blob/1a9ae5d4ab7af29a165f6b32006d7ad70768b3b4/state/annotations.go#L30  this should be called set.

Reply:
In the old code unsetting keys seems to be done via setting the value to `""`, which adds the key into a separate set called `toRemove`. I'm not a huge fan of that, but I'd say we keep the current functionality unless we have a better solution. Providing an explicit function for a functionality that virtually no-one will use feels a bit like an overkill to me.